src/pybind/mgr/cephadm/ssh.py

```diff
@@ -112,7 +112,7 @@ def redirect_log(self, host: str, addr: str) -> Iterator[None]:
         except OSError as e:
             self.mgr.offline_hosts.add(host)
             log_content = log_string.getvalue()
-            msg = f"Can't communicate with remote host `{addr}`, possibly because python3 is not installed there or you are missing NOPASSWD in sudoers. {str(e)}"
+            msg = f"Can't communicate with remote host `{addr}`, possibly because the host is not reachable or python3 is not installed on the host. {str(e)}"
```

Comment:
@phlogistonjohn I dropped it intentionally because I want the errors to be more specific. As of the changes on this PR, in case of a sudo issue you will get a clear message (as bellow). In fact, the above error only happens when the host is not reachable. When python is not installed you will get an error saying that. I tested all these scenarios on my cluster (you can find the logs for each are on the PR description). For example when sudo is not granted or not working properly you will get:

```
[ceph: root@ceph-node-0 /]# ceph orch host add ceph-node-2 192.168.100.102
Error EINVAL: Command ['which', 'python3'] failed. 
We trust you have received the usual lecture from the local System
Administrator. It usually boils down to these three things:

    #1) Respect the privacy of others.
    #2) Think before you type.
    #3) With great power comes great responsibility.

sudo: a terminal is required to read the password; either use the -S option to read from standard input or configure an askpass helper
sudo: a password is required
```